### PR TITLE
wireless/bcm43xxx: set listen interval on lowpower

### PR DESCRIPTION
**Pull request by anchao**
## Summary

wireless/bcm43xxx: set listen interval on lowpower

set listen interval dtim(Delivery Traffic Indication Message) on lowpower mode

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

N/A

## Testing

bcm43013

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c [hunk 1]
```diff
@@ -492,7 +464,10 @@ int bcmf_wl_active(FAR struct bcmf_dev_s *priv, bool active)
 
   /* Set default power save mode */
 
-  ret = bcmf_wl_set_pm(priv, PM_OFF);
+  out_len = 4;
+  value   = PM_OFF;
+  ret     = bcmf_cdc_ioctl(priv, interface, true, WLC_SET_PM,
+                           (uint8_t *)&value, &out_len);
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c:470]:
```suggestion
                           (FAR uint8_t *)&value, &out_len);
```

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c [hunk 2]
```diff
@@ -1811,3 +1786,37 @@ int bcmf_wl_get_country(FAR struct bcmf_dev_s *priv, FAR struct iwreq *iwr)
...
+    }
+
+  if (priv->lp_dtim == interval_ms)
+    {
+      return OK;
+    }
+
+  ret = bcmf_cdc_iovar_request(priv, CHIP_STA_INTERFACE, true,
+                               IOVAR_STR_LISTEN_INTERVAL_DTIM,
+                               (uint8_t *)&value, &out_len);
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c:1813]:
```suggestion
                               (FAR uint8_t *)&value, &out_len);
```

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c [hunk 3]
```diff
@@ -788,32 +799,57 @@ static int bcmf_ifdown(FAR struct net_driver_s *dev)
...
+{
+  FAR struct bcmf_dev_s *priv = arg;
+
+  if (bcmf_lowpower_expiration(arg, &priv->lp_work_ifdown,
+                               bcmf_lowpower_ifdown_work,
+                               SEC2TICK(LP_IFDOWN_TIMEOUT)))
+    {
+      if (priv->bc_bifup)
+        {
+          extern void netdev_ifdown(FAR struct net_driver_s *dev);
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c:851]: can `net/netdev/netdev.h` be somehow included instead of `extern`?

Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c]: Done

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c (hunk 2 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c]: Done

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c (hunk 1 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c]: Done

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**